lms/djangoapps/teams/views.py

```diff
@@ -877,6 +876,14 @@ def post(self, request):
         except User.DoesNotExist:
             return Response(status=status.HTTP_404_NOT_FOUND)
 
+        course_module = modulestore().get_course(team.course_id)
+        if 'max_team_size' in course_module.teams_configuration \
+                and team.users.count() >= course_module.teams_configuration['max_team_size']:
+            return Response(
+                build_api_error(ugettext_noop("The team is already full.")),
+                status=status.HTTP_400_BAD_REQUEST
+            )
```

Comment:
I wonder if it makes sense to include this check in the actual `team.add_user` method itself? It'd incur an extra performance hit whenever anyone tried to add someone to a team, but it would protect us from having to patch everything like this one level up.